UX: Settings screen improvements (tab per AI backend, split files)

acidbubbles opened the pull request:
![image](https://user-images.githubusercontent.com/34192666/230184572-1b122788-801f-407d-b525-6568c6c4f557.png)

- Move each AI backend into it's own screen
- Move presets in each AI backend's screen
- Show a loading indicator
- Show the default backend when entering the screen
- Refactor to split each backend UI into it's own file

I tried to keep the same form structure and keep changes to a minimum

acidbubbles committed:
Extract AI settings in a separate file with one tab per AI

acidbubbles committed:
Select the default AI backend by default

acidbubbles committed:
Move presets in each AI tab

acidbubbles committed:
Extract Horde Settings

acidbubbles committed:
Extract OpenAI Settings

acidbubbles committed:
Extract Scale Settings

acidbubbles committed:
Extract NovelAI Settings

acidbubbles committed:
Extract other AI backend settings, show loading

acidbubbles commented:
I have some additional suggestions;

- Split "Backends" and "Settings", so "UI Settings" becomes just "Settings" and "Backends" becomes the "AI" part of the settings. That might be simpler than having multiple levels of AI.
- I would move the large home page documentation into separate GitHub pages, with a link in each backend settings page.

I can do both changes if you want (as a separate PR or in this one, as you wish)

sceuick commented on web/pages/Settings/index.tsx:
```diff
@@ -35,22 +29,16 @@ const adapterOptions = CHAT_ADAPTERS.filter((adp) => adp !== 'default') as Defau
-  const refreshHorde = () => {
-    settingStore.getHordeModels()
-    settingStore.getHordeWorkers()
+  let hordeWorkers = state.user?.hordeWorkers || []
```
Need to use signals for any kind of state.

sceuick commented on web/pages/Settings/index.tsx:
```diff
@@ -35,22 +29,16 @@ const adapterOptions = CHAT_ADAPTERS.filter((adp) => adp !== 'default') as Defau
 
-  const tabs = ['ai', 'ui', 'service'] satisfies Tab[]
+  const tabs: Tab[] = ['ai', 'ui']
+  if(!state.loggedIn) tabs.push('guest')
```
I'd highly recommend running using the Prettier as your default formatter and configure your IDE to format on save. 
In the mean time run `pnpm format:fix` before pushing.

acidbubbles replied:
I will from now on!

sceuick commented on web/pages/Settings/AISettings.tsx:
```diff
@@ -0,0 +1,99 @@
+        </div>
+      </Show>
+
+      <div class={currentTab() === 'horde' ? tabClass : 'hidden'}>
```
This looks great on desktop, but need to consider mobile as well.

acidbubbles replied:
To me it worked well, although it needs scrolling, but it was this way before, too. If/when I work on presets and moving those in a separate Backend menu, maybe there's a different UI better suited for this, such as a list that brings you to a dedicated per-backend page instead of tabs (like the chats list that brins you to the chat details).

sceuick committed:
- Re-add cluade
- Re-add openai usage
- Change mobile title
- Enable x scroll on tabs for mobile

sceuick commented:
Great work! :heart: 
Looks fantastic.
